sdk/core/Azure.Core/api/Azure.Core.netcoreapp2.1.cs

```diff
@@ -1134,6 +1134,12 @@ public abstract partial class ObjectSerializer
         public abstract System.Threading.Tasks.ValueTask SerializeAsync(System.IO.Stream stream, object? value, System.Type inputType, System.Threading.CancellationToken cancellationToken);
         public virtual System.Threading.Tasks.ValueTask<System.BinaryData> SerializeAsync(object? value, System.Type? inputType = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
     }
+    public abstract partial class SchemaValidator<T>
+    {
+        protected SchemaValidator() { }
+        public abstract T GenerateSchema(System.Type dataType);
+        public abstract void Validate(object data, System.Type dataType, T schemaDefinition);
```

Comment:
Initially we did have a validate method called IsValid that returned a boolean. I ended up wanting to change it because in general, the available third-party libraries return a list of some kind of validation error messages. In my mind the easiest way to bubble those up from this method is to throw it in an exception.

We can change this to `public abstract bool TryValidate` and throw from the serializer as an alternative. Do you have a suggestion for a good way to preserve validation errors for the customers if we do change it?
We could also have both TryValidate and Validate and make them optional, so users can define one or the other, depending on if they want to utilize exceptions?

Reply:
I changed it back to `public abstract bool IsValid`, which I think hits both scenarios where users that want to just return a bool can, and users that want to throw an exception can do so as well (since my understanding of the try pattern is that it shouldn't throw an exception).

Reply:
@jsquire I defined it as a [method](https://github.com/Azure/azure-sdk-for-net/blob/3f0b9f7cbaa6be948eec8f54b8e439746503026e/sdk/core/Azure.Core.Experimental/src/SchemaValidator/SchemaValidator.cs#L28), but it does sound like a property which is probably a sign it's not a good name... I went with it because I think one iteration we had included a method named `IsValid`.

But for the pair of methods I was confused about how Validate calling TryValidate is able to preserve validation errors. If you wanted to do that you would have to implement TryValidate to return True, ignore it, and then do all the validation in the Validate function so you can wrap the errors in the exception. I think maybe just having two independent methods, `TryValidate` and `Validate` both virtual is a more intuitive pattern?

Reply:
If we're looking to preserve errors, then we'd return them via `out` param.    Sorry, I was looking only at the API file so I missed that we were embedding information in the exception.   Snippet above has been adjusted.